### ISSUE TITLE
OS dependency checks should be smarter

### ISSUE THREAD

**Issue by verdurin** [hpcugent/easybuild-easyconfigs#2022]
Working on my UCSC-tools easyconfig, I have an OS dependency specified for mysql-devel. On CentOS, this is met by the mariadb-devel package:

```
[centos@easybuild-6 x86_64]$ rpm -q --provides mariadb-devel
mariadb-devel = 1:5.5.44-1.el7_1
mariadb-devel(x86-64) = 1:5.5.44-1.el7_1
mysql-devel = 1:5.5.44-1.el7_1
mysql-devel(x86-64) = 1:5.5.44-1.el7_1
```

However, EB complains that the dependency is not met.
I think it should have a smarter check that takes account of this kind of thing.

**Comment by boegel** [comment]
Right now, `eb` just uses `rpm -q <depname>`, see https://github.com/hpcugent/easybuild-framework/blob/master/easybuild/tools/systemtools.py#L359

What should it be using instead?

We need something the other way around, i.e. an alternative to `rpm -q mysql-devel` (since we don't know which package provides this, i.e. we can't use `mariadb-devel`.

Alternatively, you can list both `mysql-devel` and `mariadb-devel` as options:

```
osdependencies = [('mysql-devel', 'mariadb-devel')]
```

**Comment by verdurin** [comment]
Yes, I've worked around it for now by adding both alternatives in the easyconfig.

Maybe that's all that's needed here. I suppose installing it via yum protected me from this at first.

**Comment by fgeorgatos** [comment]
Yet, the tuple operator is mostly a workaround for the majority of current
issues,
however it is not the complete answer to it:
* what about fi. Boost libs which are packaged differently across different
linux bistro families?
(yeah, you could rely on eb provided boost for that, but try to see the
more generic need)

IMHO, this suggests for a needed solution which can interface nicely with
existing CM /
Configuration Management mechanisms and even more generic things (such as
ports, macports etc).
Speaking of which, I'm inclined to ask: which CM solution have you seen to
be more linux distro family agnostic,
as regards setting up packages for a given need? (example: install package
which provides `host` command)



On 2 October 2015 at 15:50, Adam Huffman <notifications@github.com> wrote:

> Yes, I've worked around it for now by adding both alternatives in the
> easyconfig.
>
> Maybe that's all that's needed here. I suppose installing it via yum
> protected me from this at first.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-easyconfigs/issues/2022#issuecomment-145047457>
> .
>



-- 


echo "sysadmin know better bash than english"|sed s/min/mins/ \
  | sed 's/better bash/bash better/' # signal detected in a CERN forum
